demonstrations/tutorial_vqe_bond_dissociation.py

```diff
@@ -0,0 +1,534 @@
+activation energy barriers, reaction energies, and reaction rates. As illustrative examples, we
+use tools implemented in PennyLane to study diatomic bond dissociation and reactions involving 
+the exchange of hydrogen atoms. Let's get started! 
+
+
+
+.. figure:: /demonstrations/vqe_bond_dissociation/h2_pes_pictorial.png
+   :width: 90%
+   :align: center
+   
+   An illustration of the potential energy surface of bond dissociation for the hydrogen molecule. 
+   On the :math:`y`-axis is the total energy and :math:`x`-axis is the internuclear bond
+   distance. By looking at this curve, we can estimate the :math:`H-H` bond length and the energy 
+   required to break the bond.   
+
+
+##############################################################################
+
+
+Potential Energy Surfaces: Hills to die and be reborn 
+---------------------------------------------------------------------
+
+`Potential energy surfaces (PES) <https://en.wikipedia.org/wiki/Potential_energy_surface>`_
+are, in simple words, energy landscapes on which any chemical
+reaction occurs. The concept originates from the fact that nuclei are much heavier than 
+electrons and thus move much slowly, allowing us to decouple their motion. This is known as the
+`Born-Oppenheimer approximation <https://en.wikipedia.org/wiki/Born–Oppenheimer_approximation>`_. 
+We can then solve for the electronic wavefunction with
+the nuclei clamped to their respective positions. This results in separation of nuclear and
+electronic parts of the Schrödinger equation and we then only solve the electronic
+equation:
+
+.. math:: H_{el}|\Psi \rangle =  E_{el}|\Psi\rangle.
+
+Thus arises the concept of the electronic energy of the molecule, :math:`E(R)`, 
+as a function of interatomic coordinates. The potential energy surface of a 
+molecule is a 
+:math:`n`-dimensional plot of the energy with the respect to the degrees of freedom. It gives us a 
+visual tool to understand chemical reactions where stable molecules are the local minima 
+and transition states are the hill peaks to climb.
+
+To summarize, to build the potential energy surface, we solve the electronic Schrödinger
+equation for a set of positions of the nuclei, and subsequently move them in incremental steps
+to obtain the energy at other configurations. The obtained set of energies are then 
+plotted against nuclear positions.
+
+We begin with the simplest of molecules: :math:`H_2`. 
+The formation (or breaking) of the :math:`H-H` bond is also the simplest
+of all reactions. 
+
+.. math:: H_2 \rightarrow H + H.
+
+Using a minimal basis set (STO-3G), this system can be described by :math:`2` electrons in :math:`4` 
+spin-orbitals. When mapped to a qubit representation, we need a total of four qubits to represent it 
+and the Hartree-Fock (HF) ground state is  represented as :math:`|1100\rangle` where two
+energetically lowest orbitals are occupied and rest two are unccupied. To form the complete basis of
```

Comment:
```suggestion
spin-orbitals. When mapped to a qubit representation, we need a total of four qubits to represent it. 
The Hartree-Fock (HF) ground state is  represented as :math:`|1100\rangle`, where the two
lowest-energy orbitals are occupied, and the remaining two are unoccupied. To form the complete basis of
```
Also consider adding link to reference about the Hartree-Fock state.